src/textlint-core.js

```diff
@@ -93,38 +105,35 @@ export default class TextlintCore extends EventEmitter {
                 that.emit(`${ node.type }:exit`, node);
             }
         });
-        return {
-            filePath: '<text>',
-            messages: this.messages
-        };
+        let result = postProcess(this.messages, filePath);
+        if (result.filePath == null) {
+            result.filePath = `<Unkown${ext}>`;
+        }
+        return result;
+    }
+
+    /**
+     * lint plain text by registered rules.
+     * The result contains target filePath and error messages.
+     * @param {string} text
+     * @param {string} ext ext is extension. default: .txt
+     * @returns {TextLintResult}
+     */
+    lintText(text, ext = ".txt") {
```

Comment:
fmm, I want to delete `lintText(text)` and add `lintText(text, ext)`.
This method have two means: lint by text format, lint by user defined format.